curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md

```diff
@@ -308,26 +309,59 @@ const temp = console.log;
 try {
   console.log = obj => tempArr.push(obj);
 
-  // check that it correctly outputs the first item
+  // First item
   Math.random = () => 0;
-  getRandomLunch(testLunches);
+  getRandomLunch([...testLunches]);
   assert.strictEqual(tempArr[0], `Randomly selected lunch: ${testLunches[0]}`);
 
-  // second item
+  // Second item
   Math.random = () => 0.4;
-  getRandomLunch(testLunches);
+  getRandomLunch([...testLunches]);
   assert.strictEqual(tempArr[1], `Randomly selected lunch: ${testLunches[1]}`);
 
-  // third item
+  // Third item
   Math.random = () => 0.8;
-  getRandomLunch(testLunches);
+  getRandomLunch([...testLunches]);
   assert.strictEqual(tempArr[2], `Randomly selected lunch: ${testLunches[2]}`);
 
 } finally {
-  // restore Math.random
   Math.random = tempRandom;
   console.log = temp;
 }
+
+```
+
+The `getRandomLunch` function should not change the array passed to it as argument.
+
+```js
+const testLunches = ["Fish", "Fries", "Roast"];
+
+let originalLunches = [...testLunches]; 
+
+function isMutated(original, modified) {
+  return !original.every((v, i) => v === modified[i]);
+}
+
+try {
+  // First call
+  Math.random = () => 0;
+  getRandomLunch(testLunches);
+  assert.strictEqual(isMutated(originalLunches, testLunches), false);
+
+  // Second call
+  Math.random = () => 0.4;
+  getRandomLunch(testLunches);
+  assert.strictEqual(isMutated(originalLunches, testLunches), false);
+
+  // Third call
+  Math.random = () => 0.8;
+  getRandomLunch(testLunches);
+  assert.strictEqual(isMutated(originalLunches, testLunches), false);
+
+} catch (e) {
+  throw e;
```

Comment:
use the pattern of the previous test:
 - do not use a `catch` statement and do not throw, not useful errors are going to be printed to the console in the webpage
 - add a `finally` where you restore the changed `Math.random`